go-controller/pkg/ovn/pods.go

```diff
@@ -179,7 +179,8 @@ func (oc *Controller) addRoutesGatewayIP(pod *kapi.Pod, podAnnotation *util.PodA
 			otherDefaultRoute = otherDefaultRouteV6
 		}
 		var gatewayIP net.IP
-		if otherDefaultRoute || hybridOverlayExternalGW != nil {
+		hasRoutingExternalGWs := len(oc.getRoutingExternalGWs(pod.Namespace)) > 0
+		if otherDefaultRoute || (hybridOverlayExternalGW != nil && !hasRoutingExternalGWs) {
```

Comment:
you also need to check oc.getRoutingPodGWs(pod.Namespace) right?

Reply:
sure